src/solver/systemParser/converter.cpp

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2007-2024, RTE (https://www.rte-france.com)
+ * See AUTHORS.txt
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of Antares-Simulator,
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#include "antares/solver/systemParser/converter.h"
+
+#include <algorithm>
+
+#include "antares/solver/systemParser/system.h"
+#include "antares/study/system-model/system.h"
+
+using namespace Antares::Study;
+
+namespace Antares::Solver::SystemConverter
+{
+
+class ErrorWhileSplittingLibraryAndModel: public std::runtime_error
+{
+public:
+    explicit ErrorWhileSplittingLibraryAndModel(const std::string& s):
+        runtime_error("'.' not found while splitting library and model: " + s)
+    {
+    }
+};
+
+class LibraryNotFound: public std::runtime_error
+{
+public:
+    explicit LibraryNotFound(const std::string& s):
+        runtime_error("No library found with this name: " + s)
+    {
+    }
+};
+
+class ModelNotFound: public std::runtime_error
+{
+public:
+    explicit ModelNotFound(const std::string& s):
+        runtime_error("No model found with this name: " + s)
+    {
+    }
+};
+
+static std::pair<std::string, std::string> splitLibraryModelString(const std::string& s)
```

Comment:
Use a dedicated `struct` for the return type

Reply:
I think it's fine to use a pair and get var like this in this context `const auto [libraryId, modelId] = getSplit...`